tests/page_test.go

```diff
@@ -636,63 +631,100 @@ func TestPageWaitForFunction(t *testing.T) {
 	t.Run("ok_expr_poll_interval", func(t *testing.T) {
 		t.Parallel()
 
-		tb := newTestBrowser(t)
-		p := tb.NewPage(nil)
-		require.NoError(t, tb.runtime().Set("page", p))
-		var log []string
-		require.NoError(t, tb.runtime().Set("log", func(s string) { log = append(log, s) }))
+		vu, rt, log, cleanUp := startIteration(t)
+		defer cleanUp()
 
-		p.Evaluate(tb.toGojaValue(`() => {
+		_, err := rt.RunString(`
+		const page = browser.newPage();
+		page.evaluate(() => {
 			setTimeout(() => {
 				const el = document.createElement('h1');
 				el.innerHTML = 'Hello';
 				document.body.appendChild(el);
 			}, 1000);
-		}`))
+		});`)
+		require.NoError(t, err)
 
 		script := `
-			let resp = page.waitForFunction(%s, %s, %s);
+		const test = async function() {
+			let resp = await page.waitForFunction(%s, %s, %s);
 			if (resp) {
 				log('ok: '+resp.innerHTML());
 			} else {
 				log('err: '+err);
-			}`
+			}
+		}
+		test();`
 
 		s := fmt.Sprintf(script, `"document.querySelector('h1')"`, "{ polling: 100, timeout: 2000, }", "null")
-		_, err := tb.runJavaScript(s)
+		_, err = vu.TestRT.RunOnEventLoop(s)
 		require.NoError(t, err)
-		assert.Contains(t, log, "ok: Hello")
+		assert.Contains(t, *log, "ok: Hello")
 	})
 
 	t.Run("ok_func_poll_mutation", func(t *testing.T) {
 		t.Parallel()
 
-		tb := newTestBrowser(t)
-		p := tb.NewPage(nil)
-		require.NoError(t, tb.runtime().Set("page", p))
-		var log []string
-		require.NoError(t, tb.runtime().Set("log", func(s string) { log = append(log, s) }))
+		vu, rt, log, cleanUp := startIteration(t)
+		defer cleanUp()
 
-		_, err := tb.runJavaScript(`fn = () => document.querySelector('h1') !== null`)
-		require.NoError(t, err)
+		_, err := rt.RunString(`
+		fn = () => document.querySelector('h1') !== null
 
-		p.Evaluate(tb.toGojaValue(`() => {
+		const page = browser.newPage();
+		page.evaluate(() => {
 			console.log('calling setTimeout...');
 			setTimeout(() => {
 				console.log('creating element...');
 				const el = document.createElement('h1');
 				el.innerHTML = 'Hello';
 				document.body.appendChild(el);
 			}, 1000);
-		}`))
+		})`)
+		require.NoError(t, err)
+
+		script := `
+		const test = async function() {
+			let resp = await page.waitForFunction(%s, %s, %s);
+			log('ok: '+resp);
+		}
+		test();`
 
 		s := fmt.Sprintf(script, "fn", "{ polling: 'mutation', timeout: 2000, }", "null")
-		_, err = tb.runJavaScript(s)
+		_, err = vu.TestRT.RunOnEventLoop(s)
 		require.NoError(t, err)
-		assert.Contains(t, log, "ok: null")
+		assert.Contains(t, *log, "ok: null")
 	})
 }
 
+// startIteration will work with the event system to start chrome and
+// (more importantly) set browser as the mapped browser instance which will
+// force all tests that work with this to go through the mapping layer.
+// This returns a cleanup function which should be deferred.
+func startIteration(t *testing.T) (*k6test.VU, *goja.Runtime, *[]string, func()) {
```

Comment:
What about enforcing this for all tests in test browser? This might enable us to have more comprehensive integration tests. Or, maybe with or with this, both.

Reply:
Sure, this idea justifies a separate PR definitely.